src/App.js

```diff
@@ -43,35 +45,52 @@ class App extends React.Component {
       this.setState({ username: 'activity', phone: '8443343636' });
     }
 
-    this.isDisplayed()
+    this.loadDisplays();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.cart !== this.state.cart) {
+      this.loadDisplays();
+    }
+  }
+
+  loadDisplays = () => {
+    client.is_displayed_in_checkout(this.state.cart) 
       .then(res => this.setState({ display: res }))
-      .then(res => this.display(charges, 'checkout', 'text', 'small'))
+      .then(res => client.get_marketing_display(this.calcCharges(), 'checkout', 'text', 'small'))
       .then(res => this.setState({ checkout: res }))
-      .then(res => this.display(charges, 'pdp', 'text', 'special'))
+      .then(res => client.get_marketing_display(this.calcCharges(), 'pdp', 'text', 'special'))
       .then(res => this.setState({ marketing: res }))
-      .then(res => this.display(charges, 'pdp', 'button', 'small'))
+      .then(res => client.get_marketing_display(this.calcCharges(), 'pdp', 'button', 'small'))
       .then(res => this.setState({ marketing_button: res }))
       .then(res => this.getCustomer(this.state.email, 1))
       .catch(err => console.log(err));
   }
 
-  makePhoneNumber() {
-    let segment = (min, max) => (min + Math.random() * (max - min)).toFixed();
-
-    return `+1 800 ${segment(2, 9)}${segment(0, 9)}${segment(2, 9)} ${segment(1000, 1900)}`;
+  onChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
   }
 
-  getCustomer(email, customerId) {
-    return client.get_customer(email, customerId)
-      .then(res => console.log(res));
+  setCartItemPrice = e => {
+    let cartItem = this.state.cart[0];
+    cartItem.data.price = parseInt(e.target.value);
```

Comment:
looks like we lose the cents in the API request, due to this `parseInt` call?

Reply:
which API request?